docs/src/index.md

```diff
@@ -208,12 +208,9 @@ A `LazyBufferCache` is a `Dict`-like type for the caches, which automatically de
 new cache arrays on demand when they are required. The function `f` maps
 `size_of_cache = f(size(u))`, which by default creates cache arrays of the same size.
 
-Note that `LazyBufferCache` does cause a dynamic dispatch, though it is type-stable.
-This gives it a ~100ns overhead, and thus on very small problems it can reduce
-performance, but for any sufficiently sized calculation (e.g. >20 ODEs) this
-may not be even measurable. The upside of `LazyBufferCache` is that the user does
-not have to worry about potential issues with chunk sizes and such: `LazyBufferCache`
-is much easier!
+Note that `LazyBufferCache` is type-stable and contains no dynamic dispatch. This gives
```

Comment:
doesn't it have an internal dynamic dispatch?

Reply:
No, it's just a lookup from a `Dict{Any, Any}`, which is not type stable, but doesn't have dynamic dispatch either.

Verified via `@btime` indicating no allocations and via `@profview` being a happy color (dynamic dispatch is supposed to show up in red)

![Pasted image 3](https://github.com/SciML/PreallocationTools.jl/assets/60898866/d9188be4-58b8-4087-9aca-1086bfdbcd00)

(In theory the correct way to verify is JET)

Reply:
wow that's pretty incredible. Julia used to not be able to omit that.